feelslikemonday/app/src/main/java/com/example/feelslikemonday/DAO/FollowPermissionDAO.java

```diff
@@ -0,0 +1,99 @@
+package com.example.feelslikemonday.DAO;
+
+import android.util.Log;
+
+import androidx.annotation.NonNull;
+
+import com.example.feelslikemonday.model.FollowPermission;
+import com.google.android.gms.tasks.OnCompleteListener;
+import com.google.android.gms.tasks.OnFailureListener;
+import com.google.android.gms.tasks.OnSuccessListener;
+import com.google.android.gms.tasks.Task;
+import com.google.firebase.firestore.DocumentSnapshot;
+import com.google.firebase.firestore.FirebaseFirestore;
+
+import static androidx.constraintlayout.widget.Constraints.TAG;
+
+public class FollowPermissionDAO {
+    private final FirebaseFirestore db = FirebaseFirestore.getInstance();
+    private static final FollowPermissionDAO instance = new FollowPermissionDAO();
+
+    private final String COLLECTION_NAME = "followPermissions";
+
+    private FollowPermissionDAO(){}
+
+    public static FollowPermissionDAO getInstance() {
+        return instance;
+    }
+
+    public void createOrUpdate(String username, FollowPermission followPermission, final VoidCallback onSuccess){
+        db.collection(COLLECTION_NAME).document(username)
+                .set(followPermission)
+                .addOnSuccessListener(new OnSuccessListener<Void>() {
+                    @Override
+                    public void onSuccess(Void aVoid) {
+                        Log.d(TAG, "DocumentSnapshot successfully written!");
+                        onSuccess.onCallback();
+                    }
+                })
+                .addOnFailureListener(new OnFailureListener() {
+                    @Override
+                    public void onFailure(@NonNull Exception e) {
+                        Log.w(TAG, "Error writing document", e);
+                    }
+                });
+    }
+
+    public void delete(String username, final VoidCallback onSuccess){
+        db.collection(COLLECTION_NAME).document(username)
+                .delete()
+                .addOnSuccessListener(new OnSuccessListener<Void>() {
+                    @Override
+                    public void onSuccess(Void aVoid) {
+                        Log.d(TAG, "DocumentSnapshot successfully deleted!");
+                        onSuccess.onCallback();
+                    }
+                })
+                .addOnFailureListener(new OnFailureListener() {
+                    @Override
+                    public void onFailure(@NonNull Exception e) {
+                        Log.w(TAG, "Error writing document", e);
+                    }
+                });
+    }
+
+    /**
+     * Queries for a FollowPermission by the user's username. A callback method will be called on success
+     */
+    public void get(String username, final FollowPermissionCallback onSuccess){
```

Comment:
why do we have this get method?

Reply:
Just in case someone doesn't want to add an onFail callback. This might be a bad practice for me to allow, so I might remove it